### PR TITLE
Air travel

### PR DESCRIPTION
**Pull request by tannguyen153**
The new update allows travel agents and their travel paths to be justified by air travel statistical data. The code to move particles around is the same as in random travel. Both air and random travel agents share the same travel particle container (on_travel_pc) and bin (m_bins_travel).

### CODE REVIEW

**Comment by tannguyen153** [comment]
I'm thinking of multiplying the travel probability with the travel model frequency. If we run the travel model every 2 days as opposed to once per day, the number of travel agents should be 2x. This is to make sure that the annual number of travelers is consistent with the input data. The probabilities of travel path should be independent of travel model frequency.

**Comment by tannguyen153** [comment]
I just resolved the conflicts and merged  latest codes in development to the airTravel branch. This PR is no longer behind.

File: examples/inputs.ca [hunk 1]
```diff
@@ -3,10 +3,13 @@ agent.census_filename = "../../data/CensusData/CA.dat"
 agent.workerflow_filename = "../../data/CensusData/CA-wf.bin"
 agent.initial_case_type = "file"
 agent.case_filename = "../../data/CaseData/July4.cases"
+agent.airports_filename="../../data/CensusData/CA_airports.dat"
+agent.air_traffic_filename= "../../data/CA_CY23AirTraffic.dat"
 
 agent.nsteps = 120
 agent.plot_int = 10
 agent.random_travel_int = 24
+agent.air_travel_int = 1
```
Comment (debog) [inline examples/inputs.ca:12]: Can you add these to the documentation?

File: src/AgentContainer.cpp [hunk 2]
```diff
@@ -204,8 +204,6 @@ void AgentContainer::moveAgentsToWork ()
     }
 
     m_at_work = true;
-
-    Redistribute();
```
Comment (debog) [inline src/AgentContainer.cpp]: @stevenhofmeyr specifically added the `Redistribute()` in [this PR](https://github.com/AMReX-Codes/ExaEpi/pull/79). Can it really be removed?

File: src/AgentContainer.cpp [hunk 3]
```diff
@@ -251,8 +249,6 @@ void AgentContainer::moveAgentsToHome ()
     }
 
     m_at_work = false;
-
-    Redistribute();
```
Comment (debog) [inline src/AgentContainer.cpp]: @stevenhofmeyr specifically added the `Redistribute()` in [this PR](https://github.com/AMReX-Codes/ExaEpi/pull/79). Can it really be removed?

File: src/AgentContainer.cpp [hunk 4]
```diff
@@ -343,16 +480,68 @@ void AgentContainer::returnRandomTravel (const AgentContainer& on_travel_pc)
...
+                auto prob_ptr        = soa.GetRealData(r_RT+r0(d)+RealIdxDisease::prob).data();
+                auto prob_ptr_travel = soa_travel.GetRealData(r_RT+r0(d)+RealIdxDisease::prob).data();
+
+                amrex::ParallelFor( np_travel,
+                    [=] AMREX_GPU_DEVICE (int i) noexcept
+                    {
+                        int dst_index = air_travel_ptr_travel[i];
+                        if(dst_index>=0){
+                                prob_ptr[dst_index] += prob_ptr_travel[i];
+                                       air_travel_ptr[dst_index] = -1;
```
Comment (debog) [inline src/AgentContainer.cpp]: indentation?

File: src/AgentContainer.cpp [hunk 5]
```diff
@@ -343,16 +480,68 @@ void AgentContainer::returnRandomTravel (const AgentContainer& on_travel_pc)
...
+                                random_travel_ptr[dst_index] = -1;
+                        }
                     });
             }
         }
     }
 }
 
+
+/*! \brief Return agents from random travel
```
Comment (debog) [inline src/AgentContainer.cpp]: Update this comment

File: src/AgentContainer.cpp [hunk 6]
```diff
@@ -686,8 +875,18 @@ void AgentContainer::interactNight ( MultiFab& a_mask_behavior /*!< Masking beha
...
                                             AgentContainer& on_travel_pc /*< agents that are on random_travel */)
 {
-    BL_PROFILE("AgentContainer::interactNight");
+    BL_PROFILE("AgentContainer::interactRandomTravel");
     if (haveInteractionModel(ExaEpi::InteractionNames::random)) {
         m_interactions[ExaEpi::InteractionNames::random]->interactAgents( *this, a_mask_behavior, on_travel_pc);
     }
 }
+
+/*! \brief Interaction with agents on random travel */
```
Comment (debog) [inline src/AgentContainer.cpp]: Update the comment.

**Comment by debog** [comment]
@tannguyen153 I may be missing something here but it seems to me that:
+ The main difference between random travel and air travel is that the travel location is random in the former and computed from actual air travel data in the latter.
+ The actual interaction between agents that are on travel with themselves and those that reside/work there seems to be exactly the same for both these cases.

Is it possible to unify the two and reduce duplicated code? For example, rename `InteractionModRandom` to `InteractionModTravel` and have it called from both? Maybe it should be called only once because the `on_travel_pc` contains agents that are on both random and air travel, right?

File: src/AgentContainer.cpp (hunk 2 above)
Comment (tannguyen153) [reply src/AgentContainer.cpp]: Hi @stevenhofmeyr, Redistribute() operation is costly so Andrew created a small particle container for travel agents and only redistribute agents in this container. For work and home activities, we have a static communication pattern which can be handled by the arrangement of bins.

**Comment by tannguyen153** [comment]
Hi Debo, I kept the data movement and interaction code of random and air travel the same is just for now. This is like a place holder which makes the code validation easier. In future versions, air travel agents can have a different interaction behavior compared to others who take another type of transportation. Keeping a separate code will allow me to modify the interaction behavior without dealing with code conflicts.

File: src/AgentContainer.cpp (hunk 2 above)
Comment (debog) [reply src/AgentContainer.cpp]: @tannguyen153 (and @atmyers), I tagged you both on a thread in the ExaEpi channel on the EMERGE Slack - @stevenhofmeyr explained there why he add these `Redistribute()` calls. Can you take a look?

**Pull request merged**